cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -140,6 +140,14 @@ private void createLocation(NamespaceMeta namespaceMeta) throws IOException {
             "with sufficient privileges for the user and then try creating a namespace.",
           customNamespacedLocation.toString(), namespaceMeta.getNamespaceId()));
       }
+      // we also expect it to empty since non-empty directories can lead to various inconsistencies CDAP-6743
+      if (customLocation.list().length != 0) {
+        // TODO: Add username in the below exception message.
```

Comment:
why is this a TODO still? we already have this info in the ``NamespaceMeta`` right?